### PR TITLE
feat(lsp-types): allow use of proposed spec types

### PR DESCRIPTION
**Pull request by sno2**
Towards #329

### CODE REVIEW

File: examples/custom_notification.rs [hunk 1]
```diff
@@ -45,6 +45,8 @@ impl LanguageServer for Backend {
                 }),
                 ..ServerCapabilities::default()
             },
+						#[cfg(feature = "proposed")]
+            offset_encoding: None,
```
Comment (ebkalderon) [inline examples/custom_notification.rs:49]: I'm fairly sure this type implements `Default`. Any possibility we could replace these `#[cfg]` annotated fields with a `..Default::default()` initializer at the bottom instead?

**Review by ebkalderon — CHANGES_REQUESTED** [review]
This looks terrific! Thanks so much for the pull request. I agree with the described stability policy for the `proposed` feature. Just left one small piece of feedback for review before merging.

**Comment by sno2** [comment]
I've fixed the issue and hopefully I won't get another formatting error as I ran `cargo fmt` before pushing.

**Review by ebkalderon — APPROVED** [review]
Awesome, thanks so much for addressing the feedback quickly! I think this is good to merge. :rocket:

**Comment by ebkalderon** [comment]
Looks like two cases of `..Default::default()` in `examples/tcp.rs` are deemed unnecessary by Clippy, which is why CI is failing.

**Comment by sno2** [comment]
> Looks like two cases of `..Default::default()` in `examples/tcp.rs` are deemed unnecessary by Clippy, which is why CI is failing.

I think that clippy isn't being run with the `proposed` flag which is why clippy is giving an error.

**Comment by silvanshade** [comment]
Going to merge this since it seems the feedback was addressed and the clippy issue was fixed. Thanks for the PR!

**Comment by sno2** [comment]
@silvanshade Hey, thanks for pulling my changes. Do you know if you could push a version bump for this integration or a rough time when the next release will be if there's a loose schedule?

**Comment by silvanshade** [comment]
@sno2 I don't think I have the ability to publish a new version of the crate. @ebkalderon any thoughts on making a new release for this feature?

**Comment by ebkalderon** [comment]
Sure, I'm publishing the release right now! Sorry for the delay. Will reply here again once it's on Crates.io.

**Comment by ebkalderon** [comment]
Release [v0.17.0](https://github.com/ebkalderon/tower-lsp/releases/tag/v0.17.0) is now out on both GitHub and Crates.io! :tada:

**Pull request merged**